ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,448 @@
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapFloat1() returns error? {
+    var x1 = xml `<!-- outer comment -->`;
+    var x2 = xml `<name>0.5</name>`;
+    xml x3 = x1 + x2;
+
+    map<float> expected = {
+        name: 0.5
+    };
+
+    map<float> actual = check fromXml(x3);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapXml1() returns error? {
+    var x1 = xml `<name>0.5</name>`;
+
+    map<xml> expected = {
+        "#content": xml `<name>0.5</name>`
+    };
+
+    map<xml> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+xml xmlData = xml `<Invoice xmlns="example.com" attr="attr-val" xmlns:ns="ns.com" ns:attr="ns-attr-val">
+                        <PurchesedItems>
+                            <PLine><ItemCode>223345</ItemCode><Count>10</Count></PLine>
+                            <PLine><ItemCode>223300</ItemCode><Count>7</Count></PLine>
+                            <PLine><ItemCode discount="22%">200777</ItemCode><Count>7</Count></PLine>
+                        </PurchesedItems>
+                        <Address xmlns="">
+                            <StreetAddress>20, Palm grove, Colombo 3</StreetAddress>
+                            <City>Colombo</City>
+                            <Zip>00300</Zip>
+                            <Country>LK</Country>
+                        </Address>
+                    </Invoice>`;
+
+@test:Config {
+    groups: ["fromXml"]
+}
+function testToJsonWithComplexXmlElement() returns Error? {
+    map<json> j = check fromXml(xmlData);
+    map<json> expectedOutput = {
+        Invoice: {
+            PurchesedItems: {
+                PLine: [
+                    {ItemCode: "223345", Count: "10"},
+                    {ItemCode: "223300", Count: "7"},
+                    {
+                        ItemCode: {"discount": "22%", "#content": "200777"},
+                        Count: "7"
+                    }
+                ]
+            },
+            Address: {
+                StreetAddress: "20, Palm grove, Colombo 3",
+                City: "Colombo",
+                Zip: "00300",
+                Country: "LK",
+                "xmlns": ""
+            },
+            "xmlns:ns": "ns.com",
+            "xmlns": "example.com",
+            "attr": "attr-val",
+            "ns:attr": "ns-attr-val"
+        }
+    };
+    test:assertEquals(j, expectedOutput, msg = "testToJsonComplexXmlElement result incorrect");
+}
+
+type Commercial4 record {
+    BookStore4 bookstore;
+};
+
+type BookStore4 record {
+    string storeName;
+    int postalCode;
+    boolean isOpen;
+    Address4 address;
+    Codes4 codes;
+    @Attribute
+    string status;
+    @Attribute
+    string 'xmlns\:ns0;
+};
+
+type Address4 record {
```

Comment:
Don't we need this to be `Address` align with the XML parent element.

Reply:
Validated parent record name with XML root element name in [0d33892](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/0d338920c6673f4f1c8be7dfbcfd35bc3f0849dc). This is a sub-type, So we don't need to validate it. 
This is a sample OpenAPI

```
components:
  schemas:
    User:
      type: object
      properties:
        id:
          type: integer
        name:
          type: string
        contact_info:
          $ref: '#/components/schemas/ContactInfo'
    ContactInfo:
      type: object
      properties:
        email:
          type: string
          format: email
        phone:
          type: string
```
The XML format of this is 
```
<User>
   <id>0</id>
   <name>string</name>
   <contact_info>
      <email>user@example.com</email>
      <phone>string</phone>
   </contact_info>
</User>
```
Here, `contact_info` is another sub-object. That object can be defined under whatever name in the OpenAPI
In the above definition, that object name is `ContactInfo`

Reply:
Ack